cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/TranslationConfiguration.kt

```diff
@@ -453,6 +459,33 @@ private constructor(
             return this.configurePass(T::class, config)
         }
 
+        /**
+         * Sets a list of exclusion patterns using regular expressions for filtering files and
+         * directories.
+         *
+         * @param patterns List of exclusion patterns. Example:
+         * ```
+         * exclusionPatterns(listOf(Regex(".*test(s)?")))
+         * ```
+         */
+        fun exclusionPatternsByRegex(patterns: List<Regex>): Builder {
```

Comment:
Not sure if we should use two overloaded functions, e.g. “exclusionPatterns” or separate as here

Reply:
You can overload them.

Reply:
Done. As far as I unterstand, the `JvmName` annotation is required because the two functions only differ by a generic parameter and the JVM erases that parameter during compilation

Reply:
Can we have varargs instead of a list? Then they should be different and varargs might be nice to work with